### PR TITLE
hide cloud reference from SM docs

### PR DESCRIPTION
**Pull request by micheleRP**
## Description
This hides Cloud API and `rpk cloud` from Self Managed docs.

Resolves https://github.com/redpanda-data/documentation-private/issues/2704 
Review deadline:

## Page previews
[API reference](https://deploy-preview-734--redpanda-docs-preview.netlify.app/current/reference/api-reference/) 
[rpk commands](https://deploy-preview-734--redpanda-docs-preview.netlify.app/current/reference/rpk/)

## Checks

- [ ] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

**Comment by micheleRP** [comment]
@JakeSCahill I think all `rpk` commands should live (stay) in the `docs` repo, so this just hides `rpk cloud` from the side nav in Self-Managed docs. How do I get the `rpk cloud` folder to _not_ show `detect unlisted page`? I've played with `ifndef::env-cloud[]` placement but can't figure it out.

File: modules/ROOT/nav.adoc [hunk 1]
```diff
@@ -222,18 +222,6 @@
...
-***** xref:reference:rpk/rpk-cloud/rpk-cloud-auth-delete.adoc[]
-***** xref:reference:rpk/rpk-cloud/rpk-cloud-auth-list.adoc[]
-***** xref:reference:rpk/rpk-cloud/rpk-cloud-auth-use.adoc[]
-**** xref:reference:rpk/rpk-cloud/rpk-cloud-byoc.adoc[]
-***** xref:reference:rpk/rpk-cloud/rpk-cloud-byoc-install.adoc[]
-***** xref:reference:rpk/rpk-cloud/rpk-cloud-byoc-uninstall.adoc[]
-**** xref:reference:rpk/rpk-cloud/rpk-cloud-cluster.adoc[]
-***** xref:reference:rpk/rpk-cloud/rpk-cloud-cluster-select.adoc[]
-**** xref:reference:rpk/rpk-cloud/rpk-cloud-login.adoc[]
-**** xref:reference:rpk/rpk-cloud/rpk-cloud-logout.adoc[]
```
Comment (JakeSCahill) [inline modules/ROOT/nav.adoc:236]: We should also remove `*** xref:api:ROOT:cloud-api.adoc[]` from the nav tree?

**Comment by JakeSCahill** [comment]
> @JakeSCahill I think all `rpk` commands should live (stay) in the `docs` repo, so this just hides `rpk cloud` from the side nav in Self-Managed docs. How do I get the `rpk cloud` folder to _not_ show `detect unlisted page`? I've played with `ifndef::env-cloud[]` placement but can't figure it out.

Since these are now not being used at all in SM docs, we should move them to the `partials` directory. Partials won't be reported by the unlisted pages extension.

File: modules/ROOT/nav.adoc (hunk 1 above)
Comment (Deflaimun) [reply modules/ROOT/nav.adoc:236]: yes !

Comment (micheleRP) [reply modules/ROOT/nav.adoc:236]: done

**Comment by micheleRP** [comment]
> > @JakeSCahill I think all `rpk` commands should live (stay) in the `docs` repo, so this just hides `rpk cloud` from the side nav in Self-Managed docs. How do I get the `rpk cloud` folder to _not_ show `detect unlisted page`? I've played with `ifndef::env-cloud[]` placement but can't figure it out.
> 
> Since these are now not being used at all in SM docs, we should move them to the `partials` directory. Partials won't be reported by the unlisted pages extension.

@JakeSCahill I moved to partials and thought with by adding `:page-aliases:` the links from cloud-docs would still work, but I get errors. Do I need to update the xrefs in cloud-docs to point to this new location in partials? Also, how do I prevent 404s for users who land on [these old links](https://deploy-preview-734--redpanda-docs-preview.netlify.app/current/reference/rpk/rpk-cloud/rpk-cloud-auth/)?

**Comment by JakeSCahill** [comment]
> > > @JakeSCahill I think all `rpk` commands should live (stay) in the `docs` repo, so this just hides `rpk cloud` from the side nav in Self-Managed docs. How do I get the `rpk cloud` folder to _not_ show `detect unlisted page`? I've played with `ifndef::env-cloud[]` placement but can't figure it out.
> > 
> > 
> > Since these are now not being used at all in SM docs, we should move them to the `partials` directory. Partials won't be reported by the unlisted pages extension.
> 
> @JakeSCahill I moved to partials and thought with by adding `:page-aliases:` the links from cloud-docs would still work, but I get errors. Do I need to update the xrefs in cloud-docs to point to this new location in partials? Also, how do I prevent 404s for users who land on [these old links](https://deploy-preview-734--redpanda-docs-preview.netlify.app/current/reference/rpk/rpk-cloud/rpk-cloud-auth/)?

Page aliases don't work in partials because they don't get published. Yes, you'll need to update xrefs in the cloud docs.
To prevent 404s for these old links, we'll need to create some global redirects in our Netlify config. I can prepare that part: https://github.com/redpanda-data/docs-site/pull/75

**Review by JakeSCahill — APPROVED** [review]

**Pull request merged**